language/move-vm/runtime/src/runtime.rs

```diff
@@ -111,58 +113,127 @@ impl VMRuntime {
         data_store.publish_module(&module_id, module)
     }
 
-    // See Session::execute_script for what contracts to follow.
-    pub(crate) fn execute_script(
+    fn deserialize_args(&self, tys: &[Type], args: Vec<Vec<u8>>) -> PartialVMResult<Vec<Value>> {
+        // Check if the number of arguments matches the number of types.
+        // REVIEW: is `TYPE_MISMATCH` the most suitable error code here?
+        if tys.len() != args.len() {
+            return Err(
+                PartialVMError::new(StatusCode::TYPE_MISMATCH).with_message(format!(
```

Comment:
There is a `NUMBER_OF_TYPE_ARGUMENTS_MISMATCH`. Feels like it would also make sense to have a `NUMBER_OF_ARGUMENTS_MISMATCH`?